feat: new styles implemented

mcturco opened the pull request:
Currently working on a new version of the website with the new branding applied across all elements. Styles to match new brand have been added in tailwind config. In the process of making sure all typography and colors match new brand. Some small edits to spacing are being made here as well.

Also - wanted to make it known that there are going to be little changes in several files. If it makes more sense to separate these into several PRs, let me know! Just wanted to post the progress so far.

cc: @magicmatatjahu 

---------

**UPDATE:**
New styles have been implemented based on the current [Design System](https://www.figma.com/community/file/1063884118307396147) styles 🎉 

Here is a screenshot preview, but you may also view the changes via the deploy preview link below:

![Screen Shot 2022-02-07 at 5 36 36 PM](https://user-images.githubusercontent.com/60163079/152883895-30563c9e-55f3-4fe6-b28f-384d81cea307.png)

**Some things that were edited:**
- tailwind css config file settings
- New AsyncAPI logo added
- New color palette
- New font
- `Heading` component (applies tailwind classes based on `typeStyle` prop, with ability to change the heading level)
- `Paragraph` component (applies tailwind classes based on `typeStyle` prop)
- `TextLink` component (applies tailwind classes by default for inline links)
- The above 3 components have been applied throughout the website so that the typography remains consistent with the Design System

Do leave your feedback in this PR if you have it! 😄

mcturco committed:
new styles implemented (IP)

Styles to match new brand have been added in tailwind config. In the process of making sure all typography and colors match new brand.

mcturco committed:
Finish editing typography - home page

mcturco committed:
update footer links classes

mcturco committed:
more home page style edits

mcturco committed:
fix header width on blog page to match other pages

mcturco committed:
combing through docs styles

mcturco committed:
home page typography updates

mcturco committed:
fixing header styles and spacing on smaller screens

magicmatatjahu commented on components/DemoAnimation.js:
```diff
@@ -216,10 +217,13 @@ export default function DemoAnimation({ className = '' }) {
+            text="Play with it!"
+            level="h3"
+            className="mb-4"
+            />
```
I know that we don't have a integrated `eslint` but please make indentation for horizontal props:

```
            <Heading
              style="h3"
              text="Play with it!"
              level="h3"
              className="mb-4"
            />
```

React components are treated as objects and indentation in objects is desirable in JS world :)

Please fix it here and in other places.

mcturco replied:
Got it! Will update that across all files and noted for the future, thanks!

magicmatatjahu commented on components/typography/Heading.js:
```diff
@@ -0,0 +1,27 @@
+        </Tag>
+    )
+
+}
```
I will describe some tips, because probably it's your first React component :)

If you wanna pass something as child to the component (in your case `text`) it should be treated as children. I know that we are passing text/content etc in other component as props, but we should start moving to the `children` approach. What I am talking about? You can pass content by prop, or by children (content between component) like:

```js
<Heading>
  Some text <- this is React children
</Heading>
```

so I would like to change `text` to the `children` prop - you can check that simple blog post https://www.bigbinary.com/react-best-practices/react-children :)

Also `level` prop should be renamed to the `as` because as I see in some places, you used that component also with paragraph (`<Heading level="p"> ...`), `as` is more verbose and it indicates that you can use also another HTML tags not only headings. And of course `level` default value should be changed to the `h2` not `2` :)

Tbh I also don't like `style` prop's name, because it is similar to the `styles` prop which handles CSS styles. Maybe `styleFor`? 🤔  Also we can remove `weight` and `textColor` props and move default values to the `className` but we have to remember that we should always pass classes for weight and text color... 🤔  We can stay with `weight` and `textColor` in that case.

If you have concerns/question, feel free to raise them! :)

mcturco replied:
Thank you so much for the tips!!! And yes, this is my first "real" React component 😅  The reason for it is I saw a lot of inconsistencies with the heading treatments on the website and I wanted to make a way where it can be easy to get the right heading treatment style without having to copy and paste the class names each time.

So now that I am thinking about it, I think what I should do is just define the different levels of typography and what class names are to be rendered with each (this would go into the brand guidelines as well) and then create "identifiers" for each one, for example if I wanted to output the largest heading, the identifier might be `heading-xl` and on the `<Heading />` component, we would pass this identifier, `heading-xl`, into the current `style` prop. That way, we can remove `weight` as I don't think we should have to change this often, but I think we need to keep `textColor` as this will change often depending on the application of the heading. Unless we create identifiers for each color that would be used 🤔  but that might create some bottlenecks. Does that make sense? If not, I can work this idea into the component and we can discuss after!

> so I would like to change text to the children prop

Sounds good, I will do that 👍 

> Also `level` prop should be renamed to the `as` because as I see in some places, you used that component also with paragraph

I'm not sure that I understand the meaning of `as` in this context, but if you think it is better I am fine with updating it to that!

> Tbh I also don't like style prop's name, because it is similar to the styles prop which handles CSS styles

This is related to what I was saying above. I agree that `style` is not a great label. Would `typeStyle` make more sense here? Or maybe `treatment`?

magicmatatjahu replied:
"Identifies" for common styles classes/styles like `heading-xl` etc are a great idea! Also about your concerns about `bottlenecks`: we can create such an identifiers and then we can use `className` to override some classes, so it shouldn't be a problem. However identifiers for colors isn't good idea (for that we can use `textColor` prop or `className`).

> I'm not sure that I understand the meaning of as in this context, but if you think it is better I am fine with updating it to that!

`as` means that you wanna render root component of given React component (first HTML tag that occurs in the output of component) "as" given HTML  tag. Maybe for `heading` isn't good but for other case it should be better name of prop:

```js
function Block({ as = 'div' }) {
  const As = as

  return <As ...>
}
```

> This is related to what I was saying above. I agree that style is not a great label. Would typeStyle make more sense here? Or maybe treatment?

Yeah, `typeStyle` is better, that same as `treatment`, but maybe `identifier`? Or `preset`? :)

mcturco replied:
Okay the `as` makes more sense to me now! I think I have a new plan of creating a separate component `Paragraph` for body copy. Then in the future maybe we can create components for lists and other text markup, just to keep the class names a little bit more streamlined! So for `Heading` we can keep the `level` prop as I think it makes more sense if it were only h tags, what do you think?

And I think `typeStyle` might best define what this prop does, as `identifier`, `treatment`, and `preset` might be too vague. What do you think about that?

mcturco replied:
Just added the `Paragraph` component which could resolve the issues you mentioned here. Let me know if that works! 

A question I had as I was going through the code: In some areas, the props on the component are not stacked when being used, but instead they are inline like: `className="text-gray-800" showSupporterBanner="false"` is this treatment okay? Or should we add returns for each prop?

magicmatatjahu replied:
> Okay the as makes more sense to me now! I think I have a new plan of creating a separate component Paragraph for body copy. Then in the future maybe we can create components for lists and other text markup, just to keep the class names a little bit more streamlined! So for Heading we can keep the level prop as I think it makes more sense if it were only h tags, what do you think?

Yeah, we should think about UI library and then create a bigger components to render e.g. channels of AsyncAPI spec, single server etc :) We should built them on that simple like `Heading`, `Paragraph` etc.

> And I think typeStyle might best define what this prop does, as identifier, treatment, and preset might be too vague. What do you think about that?

go ahead with `typeStyle` :) 

> Just added the Paragraph component which could resolve the issues you mentioned here. Let me know if that works!

I will look :)

> A question I had as I was going through the code: In some areas, the props on the component are not stacked when being used, but instead they are inline like: className="text-gray-800" showSupporterBanner="false" is this treatment okay? Or should we add returns for each prop?

Sorry, but I don't understand that `...are not stacked when being used...` and `...Or should we add returns for each prop...` 😅  Do you have in mind that some component is written like:

```js
<Component prop1='...' prop2='...'>...</Component> 
```

and it should be as:

```
<Component 
  prop1='...' 
  prop2='...'
>
  ...
</Component> 
```

? Or better will be to paste some link to see that inlined component :)

mcturco replied:
@magicmatatjahu yep exactly like your example, just wondering if there is a "right" way to write out the props. In some places I realized I didn't add returns, like here:

```
<Heading level="h5" typeStyle="heading-md-semibold">
    Join our Slack workspace
</Heading>
```

instead of adding returns:

```
<Heading 
  level="h5"
  typeStyle="heading-md-semibold"
>
    Join our Slack workspace
</Heading>
```

Just wondering which format is preferred.

mcturco replied:
> Yeah, we should think about UI library and then create a bigger components to render e.g. channels of AsyncAPI spec, single server etc :) We should built them on that simple like Heading, Paragraph etc.

Is there an example of this on the website? `channels of AsyncAPI spec` `single server` Or are these things in the Studio HTML template?

magicmatatjahu replied:
If you write component like in second case you should remember about indentation, but first and second are fine :)

magicmatatjahu replied:
It's not on website but we should have a UI library for such things :) Please read that issue https://github.com/asyncapi/shape-up-process/issues/81

magicmatatjahu commented on tailwind.config.js:
```diff
@@ -27,7 +70,11 @@ module.exports = {
     colors: {
       ...colors,
-      primary: getColors(colorPrimary, [100, 200, 300, 400, 500, 600, 700, 800, 900]),
+      primary: getColors(colorPrimary, [100, 200, 300, 400, 500, 600, 700, 800, 900]), 
```
as I see `getColors` only copies the object, so I think that we can change it to the:

```suggestion
      primary: colorPrimary,
```

magicmatatjahu commented on tailwind.config.js:
```diff
@@ -3,17 +3,60 @@ const colors = require('@tailwindcss/ui/colors')
 const plugin = require('tailwindcss/plugin')
 
 const colorPrimary = {
```
Should we go only with `100, 200, 300, 500, 600` shades? Why not from 100 to 900? 🤔  I have also that same question for `colorSecondary`.

mcturco replied:
I do need to tweak the shades still. The goal was to only have 5-6 shades for primary and secondary since they are a bit brighter and more specific to the brand, whereas for `gray` you want to have at least 10 shades so that you have a wide variety of background colors and text colors. I will probably include `100, 200, 300, 400, 500, 600` shades for both `primary` and `secondary`. Do you think this is okay?

mcturco replied:
The shades for both primary and secondary have been tweaked to include 100 - 600. Let me know if we can resolve this!

magicmatatjahu replied:
> Do you think this is okay?

Yeah, it will be :)

 > The shades for both primary and secondary have been tweaked to include 100 - 600. Let me know if we can resolve this!
 
 I will look at this :)

magicmatatjahu replied:
Ok, awesome! :)

magicmatatjahu reviewed (changes requested):
I know that it's a draft PR but I see some problems.

I checked live preview and it's amazing 🚀  I am not sure but everything looks almost finished 😅  In the meantime, are you also working on a page for style guides (I ask because of the PR description)?

mcturco commented:
> In the meantime, are you also working on a page for style guides (I ask because of the PR description)?

@magicmatatjahu I have raised an issue for this so someone can help me set up the template. I tried doing it myself but I couldn't get it to work right. Here is the issue: https://github.com/asyncapi/website/issues/532

I think @yashsehgal is wanting to help out with it!

mcturco committed:
refactor Heading component

mcturco committed:
Add Paragraph component and implement on home page

mcturco committed:
Adjust primary and secondary colors

magicmatatjahu commented:
@mcturco 

> @magicmatatjahu I have raised an issue for this so someone can help me set up the template. I tried doing it myself but I couldn't get it to work right. Here is the issue: #532
>
> I think @yashsehgal is wanting to help out with it!

If you need help I can show you how to setup new page on video call :) When do you have time? Of course if @yashsehgal  don't have problem with second guy with help 😅

mcturco commented:
> If you need help I can show you how to setup new page on video call :) 

@magicmatatjahu yes! I would love that. I think @yashsehgal mentioned he was having issues with it [on the issue I raised](https://github.com/asyncapi/website/issues/532#issuecomment-1007676375)? 

Perhaps you can help us both 😄

magicmatatjahu commented on components/typography/Heading.js:
```diff
@@ -0,0 +1,42 @@
+    className,
+    children
+}) {
+    let classNames
```
Please default that var to empty string:

```suggestion
    let classNames = '';
```

Without that if someone will pass not supported `typeStyle` final css class will be render as `<h1 class="... undefined">` - undefined will be render as "undefined" :)

mcturco replied:
Good call! Just committed that suggestion for both :)

magicmatatjahu commented on components/typography/Heading.js:
```diff
@@ -0,0 +1,42 @@
+    let classNames
+    const Tag = `${level}`;
+
+    if (typeStyle === 'heading-xl') {
```
Can I have question? What is a difference between `heading` and `body`? I mean e.g. `heading-lg` and `body-lg`?

mcturco replied:
Sure, `heading` is for heading type treatments and in most cases have a heavier weight and darker color than the paragraphs, which are called `body` to represent body copy. Here is a visual breakdown of these styles and the class names assigned to each. 

![WebsiteTypography](https://user-images.githubusercontent.com/60163079/148969637-cc29877e-1a4c-401a-8526-00e310d7dcbd.jpg)

The names in the `bubble` are the identifiers which will be passed into `typeStyle` within each component and the classnames below that identifier are the tailwind classnames that are assigned to each identifier.

magicmatatjahu replied:
Ok, thanks! :) However, can we use here `switch cases` syntax not `if else`?

magicmatatjahu commented on components/typography/Paragraph.js:
```diff
@@ -0,0 +1,23 @@
+    children
+}) {
+
+    let classNames
```
That same situation as in `Heading` component, please default that var to the `""`.

mcturco replied:
Done!

magicmatatjahu reviewed (commented):
Three more comments :)

mcturco committed:
refactoring suggestion for type components

mcturco committed:
Add TextLink typography component for consistent inline link styling

mcturco committed:
Style edits to menu items, re-format logo SVG to JSX

mcturco committed:
Update to color palette based on feedback of too much pink

mcturco committed:
New studio button style

mcturco committed:
streamline heading treatment labels

mcturco committed:
Merge branch 'master' into missy-new-styles

mcturco committed:
Edits to new section styles from merge

mcturco commented:
Hi @magicmatatjahu 👋 

I think this is about ready for review. There might be some small things that I might have missed but IMO I don't think we should wait much longer to merge these changes, for things like that we can just open separate PRs. 

Initially, I thought it would be a good idea to merge this PR at the same time as #539 (brand guidelines pages), but honestly after thinking about it I don't see why the making of the brand guidelines pages have to hold up the release of this PR, but wondering what others think.

cc: @fmvilas @derberg

mcturco committed:
Load Work Sans font from Google

akshatnema commented:
Hey @mcturco sorry for intervening in the PR :sweat_smile:, as I am not much related to the PR and issue. Here are some of the changes I think you should need to consider:

1. **Hovering the links in the Footer makes the text invisible** - Maybe it's because, on hover, you have provided purple color to the text and on the black background, it's not visible. Here is my preview:
![Screenshot from 2022-02-08 10-57-42](https://user-images.githubusercontent.com/76521428/152924844-3640dd12-00b5-4734-9a5c-77c7f3a0d5a8.png)

`About` Link got disappeared (because of color) when I put my cursor on it. The same happened with the link at the footer, specifying the trademark policies.

2. **Responsive issue in Navbar in a window size of 14 inches** - If you open the dropdown menu of `Community` in Navbar and make your screen size of `1330px * 698px`, you will see that the dropdown menu is not perfectly visible in the whole screen. You probably can assign the height in percentage or just decrease some padding/margin between the options so that it gets corrected. Here's the preview:
![Screenshot from 2022-02-08 11-01-34](https://user-images.githubusercontent.com/76521428/152925798-6b135985-6c2a-42e8-9851-98f77b311fb0.png)

mcturco committed:
Adjust footer links classes

mcturco committed:
Add secondary colors to OpenInStudioButton and navigation

mcturco commented:
Hi @akshatnema 👋  No need to apologize for providing good feedback! 😄 

> Hovering the links in the Footer makes the text invisible

Good call! I think something got weird here when I merged your changes from master where you added the Link component. I think some classes got lost somehow when I was manually merging changes. Should be all good now!

> Responsive issue in Navbar in a window size of 14 inches - If you open the dropdown menu of Community in Navbar and make your screen size of 1330px * 698px, you will see that the dropdown menu is not perfectly visible in the whole screen.

Hmmm yeah I see what you mean 🤔  however, it looks like the problem could be there on master as well, although I see the problem is not that bad. I'm thinking for this we should raise a separate issue and then see who would be willing to pick it up and make the layout of that dropdown maybe match the layout of the `Learning` dropdown.

Also, I just wanted to point out that I changed the style of the `OpenInStudioButton` component to match the secondary button styling that @yashsehgal created in the Design System file. I think it is looking great paired with our primary purple, but if anyone has feedback on that I would love to hear it!

mcturco committed:
Update favicons to new brand mark

magicmatatjahu commented:
> Hmmm yeah I see what you mean 🤔 however, it looks like the problem could be there on master as well, although I see the problem is not that bad. I'm thinking for this we should raise a separate issue and then see who would be willing to pick it up and make the layout of that dropdown maybe match the layout of the Learning dropdown.

Yeah, we should make it as followup. Please don't fix bugs that we have on master, because that PR should only change styling :)  

> Also, I just wanted to point out that I changed the style of the OpenInStudioButton component to match the secondary button styling that @yashsehgal created in the Design System file. I think it is looking great paired with our primary purple, but if anyone has feedback on that I would love to hear it!

I love it ❤️

magicmatatjahu commented:
Everything looks awesome but I from my POV we have some problems:

- Some of the text has very bright contrast and I think people who have a vision problem may have trouble reading, like on Job's page:

  ![image](https://user-images.githubusercontent.com/20404945/153239963-4d2018d4-fe5d-4d8d-ae87-9858cd5336df.png)

  Current implementation has better contract in my opinion:
  
  ![image](https://user-images.githubusercontent.com/20404945/153240349-70383551-8666-4ec7-be5a-c44e973674f4.png)

- Maybe it's just me, but isn't the current font "frayed"? It seems to me that the current one that still exists has smoother edges. Maybe I'm delusional, but it seems more angular 😆 

- Testimonials area seems "empty" to me. I mean there is too much empty space which is overwhelming. In the current solution we also have a lot of gaps, but they are less "heavy" on the eye. 

  ![image](https://user-images.githubusercontent.com/20404945/153240863-ee37f075-ea06-4d74-bc26-2c36146399e4.png)
  
  ![image](https://user-images.githubusercontent.com/20404945/153241292-2c840d72-3690-4343-8be4-15f0d1c3bd73.png)

  and tbh I like more that centered align of text (not left align)
  
- I more like the centered content (but it's not a problem) in navigation (not align to right), but I understand that you wanna have consistency :) Lets see what other people think.

  ![image](https://user-images.githubusercontent.com/20404945/153241929-836beef2-b18b-4506-a007-f5c3b1737871.png)

  ![image](https://user-images.githubusercontent.com/20404945/153241842-d8841a40-832a-431e-bf17-18cc093133e9.png)
  
- we have blue arrows in the docs suggestion (I guess they should be pink/magenta?):

  ![image](https://user-images.githubusercontent.com/20404945/153242785-626d3875-0db9-47d9-92b2-a4faa1a53a4d.png)
  
- is it correct color in docs? https://deploy-preview-534--asyncapi-website.netlify.app/docs/getting-started/servers

  ![image](https://user-images.githubusercontent.com/20404945/153243376-76193a2b-fd32-4de6-bac6-21174ba49081.png)

- we have to remember that we have also `roadmap` page https://deploy-preview-534--asyncapi-website.netlify.app/roadmap I know that there's problem to render that page but we need to check if everything is rendered correctly 😄  I can help you with that, because there's hack.

mcturco committed:
jobs page contrast issues/convert tags to components

mcturco committed:
adjust type styles on tooling pages, use type components

mcturco committed:
edit blog typography using type components

akshatnema commented:
>we have to remember that we have also roadmap page [deploy-preview-534--asyncapi-website.netlify.app/roadmap](https://deploy-preview-534--asyncapi-website.netlify.app/roadmap) I know that there's problem to render that page but we need to check if everything is rendered correctly smile I can help you with that, because there's hack.

Can you please tell your hack? It is sometimes needed to me when I make some changes on `roadmap.js` :sweat_smile:.

mcturco commented:
@magicmatatjahu Hi! 👋  Thanks for your feedback
My comments are below!

> Some of the text has very bright contrast and I think people who have a vision problem may have trouble reading, like on Job's page:

Oops! I must have missed that page when I was going through updating styles using the new typography components. I also missed all of the tooling pages, so I just pushed those commits. One thing with the Modelina page... something weird is happening with the VS Code editor... Can you take a look at that for me?

> Maybe it's just me, but isn't the current font "frayed"? It seems to me that the current one that still exists has smoother edges. Maybe I'm delusional, but it seems more angular 😆

Yeah it might be a little weird to look at... Maybe we use the `Work Sans` font only for headings and then use `Inter` for body copy?

> Testimonials area seems "empty" to me. I mean there is too much empty space which is overwhelming. In the current solution we also have a lot of gaps, but they are less "heavy" on the eye.

Yeah I think this section in general needs to be improved. Maybe it seems more heavy though because of the overuse of the purple? I can adjust that to the secondary blue color instead and that might keep it from feeling too weird. The centered content paired with the left-aligned "author" area seems weird visually IMO. But like I said, this section I think needs to be rethought anyways, so maybe we don't focus too much effort on this section at this stage.

> I more like the centered content (but it's not a problem) in navigation (not align to right), but I understand that you wanna have consistency :) Lets see what other people think.

Ah yes I see what you mean. Hmmm I wonder how we can keep it consistent without it seeming like it is out of alignment 🤔 

> we have blue arrows in the docs suggestion (I guess they should be pink/magenta?):

The blue color is the correct secondary color that you also see in the Open in Studio button on the homepage 😄  I am trying to stay conscious of Fran's comment before of "too much pink" 😆  So maybe we should make all links this blue color so it is consistent?

> is it correct color in docs? https://deploy-preview-534--asyncapi-website.netlify.app/docs/getting-started/servers

This is the color that the box is currently on the website, but i agree it should probably change to the secondary blue to match

> we have to remember that we have also roadmap page https://deploy-preview-534--asyncapi-website.netlify.app/roadmap I know that there's problem to render that page but we need to check if everything is rendered correctly 😄 I can help you with that, because there's hack.

Yes! If you can help me make this visible locally that would be awesome 😄

mcturco committed:
adjust Remember component styles

magicmatatjahu commented:
@mcturco 

> Oops! I must have missed that page when I was going through updating styles using the new typography components. I also missed all of the tooling pages, so I just pushed those commits. One thing with the Modelina page... something weird is happening with the VS Code editor... Can you take a look at that for me?

Thanks! It looks better now :) For that modelina page and VSC. I don't see any "weird" things. Do you mean that old styling with dark/turquoise colors?

> Yeah it might be a little weird to look at... Maybe we use the Work Sans font only for headings and then use Inter for body copy?

If it will look better and more "smooth", we can change :)

> Yeah I think this section in general needs to be improved. Maybe it seems more heavy though because of the overuse of the purple? I can adjust that to the secondary blue color instead and that might keep it from feeling too weird. The centered content paired with the left-aligned "author" area seems weird visually IMO. But like I said, this section I think needs to be rethought anyways, so maybe we don't focus too much effort on this section at this stage.

Yeah sure, we can focus on other parts. I wrote about my first feeling on that :) Of course that section should be reworked and we should enable the passing the testimonials from community (by PRs) :)

> Ah yes I see what you mean. Hmmm I wonder how we can keep it consistent without it seeming like it is out of alignment 🤔

This is ok, but for me the current one (centered) is better :) Such my preference. We don't have to change it as others will find it better :)

> The blue color is the correct secondary color that you also see in the Open in Studio button on the homepage 😄 I am trying to stay conscious of Fran's comment before of "too much pink" 😆 

Hah ok 😆  I thought that this section didn't change, so then it's ok for me :)  

> So maybe we should make all links this blue color so it is consistent?

Hmm 🤔  I think there will be too much blue after that 😆  Lets stay with that blue, it's ok for me.

> This is the color that the box is currently on the website, but i agree it should probably change to the secondary blue to match

Yeah, Lets change it :)

> Yes! If you can help me make this visible locally that would be awesome 😄

@akshatnema @mcturco Ok, here https://github.com/asyncapi/website/blob/master/scripts/build-roadmap.js#L48 we need to have `GITHUB_TOKEN ` environment variable:
- you need to generate your own personal Github token https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token
- then you need to run the `npm run dev` by this token. If you are using MacOS or Linux then you need to run `GITHUB_TOKEN={that token} npm run dev` and you should see the rendered `roadmap` page :)

Let me know if you will have problems!

mcturco committed:
Update body font to Inter and adjust

mcturco commented:
@magicmatatjahu 

> For that modelina page and VSC. I don't see any "weird" things. Do you mean that old styling with dark/turquoise colors?

Weird I don't see the error anymore. Maybe I should just pretend like I didn't see it? 😆 

> If it will look better and more "smooth", we can change :)

Just pushed a commit where I updated the fonts to Inter for body copy, let me know what you think about it! I definitely think that it helps it look more smooth as you are saying

> This is the color that the box is currently on the website, but i agree it should probably change to the secondary blue to match --> Yeah, Lets change it :)

Updated!

And cool, I figured the hack was by using the GH token. I will try that right now and then update any styling on that page 👍  Thank you!

mcturco commented:
@magicmatatjahu I seem to still be having problems seeing the roadmap page even after using the GH token. 

Below is my command line output, I scratched out my token for safety (Lukasz told me not to share it with anyone, even my grandmother 🤷‍♀️ )
![Screen Shot 2022-02-09 at 4 45 44 PM](https://user-images.githubusercontent.com/60163079/153295717-288ab7de-2ed0-4a2d-8697-8ad2a2126963.png)

mcturco committed:
Use heading component in docs layout title

mcturco committed:
Update styles on roadmap page

mcturco commented:
@magicmatatjahu Okay! I think we might be close to being all set here. I just edited the roadmap page, so if you would like to review it that would be awesome! And let me know if we can resolve the review conversations above or if there are still actions for me to take.

magicmatatjahu commented:
@mcturco Awesome work Missy! Thanks for updating roadmap! :)

This may sound like a "rude" accusation, but in the current style of the `/job` page the rest content like the RSS information is has smaller font. Can we make them a little smaller?

Old: 
![image](https://user-images.githubusercontent.com/20404945/154703732-30c09d9a-7507-4c5b-b60a-074a200c4978.png)

New:
![image](https://user-images.githubusercontent.com/20404945/154703175-453e5b8b-881c-4ea7-9279-880784000204.png)

We should handle that in separate PR but I think that in new styles we don't have consistent colors for code styling. With old styles it fits because blue is the dominant color, but now that we have the pink color maybe we should change the colors for the code as well? What do you think about it? Or rather we should stay with that?

Old:
![image](https://user-images.githubusercontent.com/20404945/154704851-b79909e1-5bca-4f5c-b723-80d7709d15ca.png)

New:
![image](https://user-images.githubusercontent.com/20404945/154704928-17858565-3982-460d-b6cc-aaf503c61d6b.png)

Thanks for changing fonts, current it looks more smooth! However the logo is more frayed 😆  As I see it's `svg` so maybe we should improve it?

![image](https://user-images.githubusercontent.com/20404945/154705161-16e34b17-0fc2-425b-b405-c8e7986bd20a.png)

And last "problem" that I found. Did you consider to change the colors here for another one or you wanna make it in the followup PR or maybe stay with current ones?

![image](https://user-images.githubusercontent.com/20404945/154705600-5560c9c8-836d-4a6c-bf8b-64f72b56d72f.png)

Rest looks good for me!

I think that Missy needs more feedback @fmvilas @alequetzalli @boyney123 :)

magicmatatjahu commented:
@mcturco And of course please rebase with upstream master because you have a lot conflicts.

magicmatatjahu commented on components/typography/TextLink.js:
```diff
@@ -0,0 +1,20 @@
+
+export default function TextLink({
```
I prefer to name it `Href` or `Link` component :)

mcturco replied:
There is already a `Link` component I think in master? And this is for applying the `rel="noopener noreferrer"` I believe?

magicmatatjahu replied:
Yeah, also NextJS has `Link` component to make navigation between pages. Ok I see explanation about that name in other comment, we can resolve that discussion :)

magicmatatjahu commented on components/typography/TextLink.js:
```diff
@@ -0,0 +1,20 @@
+export default function TextLink({
+    href,
+    className,
+    target = '_self',
```
`_self` is default. If we pass `undefined` to the `target` react will skip that attribute, so you can change it to the:

```suggestion
    target,
```

:)

magicmatatjahu commented on components/typography/TextLink.js:
```diff
@@ -0,0 +1,20 @@
+
+    return(
+        <>
+        {' '}
```
Why we need that empty string? 😄

mcturco replied:
This is to add a space before the link because it is an inline block. If we want to remove that empty space string from the component we will just have to remember to add a space before its inclusion within text.

magicmatatjahu replied:
Ok thanks for explanation but I don't think so it's good approach. What if someone want use that link when isn't any pre content? We will add empty, unnecessary space.

🤔  Or just we should add that empty space but only when given prop is enables like `emptyPreSpace={true}`.

mcturco replied:
The idea for the component was to provide a universal way of applying a link within paragraph text. I don't really see this style being used outside of a block element, this is why I named it `TextLink` because it is meant to be applied within the text.

magicmatatjahu replied:
I may be stubborn but I don't like it 😅  I've never seen anything like it and I don't think it's necessary. We can leave it but it will have to be improved :)

magicmatatjahu reviewed (commented):
And review for code. Thanks for explanation :)

mcturco committed:
Edit jobs page paragraph styles

mcturco committed:
Update components/typography/TextLink.js

Co-authored-by: Maciej Urbańczyk <urbanczyk.maciej.95@gmail.com>

mcturco commented:
@magicmatatjahu 

> This may sound like a "rude" accusation, but in the current style of the /job page the rest content like the RSS information is has smaller font. Can we make them a little smaller?

No its not rude at all! I have committed your suggestion, let me know in the preview if it looks better

> We should handle that in separate PR but I think that in new styles we don't have consistent colors for code styling. With old styles it fits because blue is the dominant color, but now that we have the pink color maybe we should change the colors for the code as well? What do you think about it? Or rather we should stay with that?

I agree, I think we can improve this in a separate PR. I think for now its fine but we can definitely re-think the syntax highlighting to match the new brand styles.

> Thanks for changing fonts, current it looks more smooth! However the logo is more frayed 😆 As I see it's svg so maybe we should improve it?

Hmmm 🤔  I see what you mean but honestly I couldn't tell you what is causing that to happen. It is just an SVG with paths

> And last "problem" that I found. Did you consider to change the colors here for another one or you wanna make it in the followup PR or maybe stay with current ones?

I just left the colors as they are as they serve the purpose of grouping items together, so that doesn't have to necessarily be the same colors as the new brand colors.

> Ok, thanks! :) However, can we use here switch cases syntax not if else?

For some reason I couldn't reply in a comment to this review comment, but sure whatever you think it should be. Can you insert your suggestion to what it should be? I don't know how to implement switch cases 😅

mcturco committed:
Merge branch 'master' into missy-new-styles

magicmatatjahu commented:
@mcturco About that `switch cases` block:

Currently you wrote everything with `if else`, but if you have few `else if` better is to use the `switch case`. Consider change:

- current:

```js
if (typeStyle === 'heading-xl') {
  ...
} else if (typeStyle === 'heading-lg') {
  ...
} else if (typeStyle === 'heading-md') {
  ...
}
```

- `switch case`:

```js
let classNames = '';
switch (typeStyle) {
  case 'heading-xl': {
    classNames = `font-heading text-2xl font-bold tracking-heading md:text-6xl leading-snug ${className || ''}`;
    break;
  };
  case 'heading-lg': {
    classNames = `font-heading text-2xl font-bold tracking-heading md:text-4xl leading-snug ${className || ''}`
    break;
  };
  case 'heading-md': {
    classNames = `font-heading text-2xl font-bold tracking-heading leading-snug ${className || ''}`
    break;
  };
  ...
}
```

Of course `if else if` is also good, but `switch case` usually is more readable :)

mcturco committed:
Fixing errors after merging

mcturco committed:
Refactor typography components using switch cases

mcturco commented:
@magicmatatjahu Thank you for the tips!

Check now and see if that's better 😄

magicmatatjahu commented on components/typography/Heading.js:
```diff
@@ -0,0 +1,53 @@
+    switch (typeStyle) {
+        case 'heading-xl': {
+            classNames = `font-heading text-2xl font-bold tracking-heading md:text-6xl leading-snug ${className || ''}`
+        }
```
When you use `switch case` you need to remember about `break` statement. Here you have info why it's so important :) 

https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Statements/switch#what_happens_if_i_forgot_a_break

mcturco replied:
Gotcha. Do I need `default` there at the end of the last case? If so, how should I handle that?

magicmatatjahu replied:
Here you have `typeStyle` set to `heading-lg` default so you can write:

```js
default: {
  classNames = `font-heading text-2xl font-bold tracking-heading md:text-4xl leading-snug ${className || ''}`;
}
```

at last case in the `switch`. Here you don't need break because you call this as last case :)

mcturco replied:
Ok check now

magicmatatjahu commented on components/typography/Paragraph.js:
```diff
@@ -0,0 +1,26 @@
+    switch (typeStyle) {
+        case 'body-lg': {
+            classNames = `text-lg leading-relaxed font-regular ${className || ''}`
+        }
```
When you use switch case you need to remember about break statement. Here you have info why it's so important :)

https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Statements/switch#what_happens_if_i_forgot_a_break

magicmatatjahu replied:
Here you should make:

```js
default: { 
  classNames = `text-lg leading-relaxed font-regular ${className || ''}`
}
```

mcturco replied:
Ok check now

magicmatatjahu reviewed (commented):
Two comments about `switch case` :)

magicmatatjahu commented on components/navigation/BlogPostItem.js:
```diff
@@ -22,24 +24,65 @@ export default function BlogPostItem({ post, className = '' }) {
+              <span>
+                {post.readingTime} min read
+              </span>
+            </Paragraph>
```
You should remove these lines and everything should works + of course add that `break` :)

```suggestion
```

mcturco replied:
On no!! Something weird must have happened when I merged the changes. My bad 😓

mcturco replied:
Ok check now

mcturco committed:
Bug fixes

magicmatatjahu commented:
@mcturco 👌🏼  Lets see what other people will think :) Could you write about that PR on Slack as we discussed?

akshatnema commented:
Hey @mcturco, I saw the preview right now. Really, great work done by you in typography, styling, and design. You nailed it ✨. Overall, PR looks top-notch and fully complete. Just look like a few finishing touches are needed and then, this PR will boom the website 💥. Here are some suggestions from my side:

- ### URL Text Link not visible on hover
![Screenshot (25)_LI](https://user-images.githubusercontent.com/76521428/155689902-fac46186-677f-40c5-a18e-9285c36ec20f.jpg)

Yep, I completely understand that you have made a default styling for all the links on the website, but I think we have to make the hover effect of this link the same as that of `About`, `Blog`, etc. in the footer.


- ### A Very Fine Line Gap on top of Navbar
![Screenshot 2022-02-25 150015](https://user-images.githubusercontent.com/76521428/155691016-887dea79-d761-4f94-a9ee-9279ceb0af91.png)

This error is still on the website right now as well. Looks like there is a very fine gap between the top and navbar of the website. I don't know why it is there and how to resolve it. I will just suggest, if you find this error, kindly resolve it 😄. 


- ### Extra indentation for title in docs
![Screenshot 2022-02-25 151126](https://user-images.githubusercontent.com/76521428/155692404-e46036e3-45f4-4e29-abf4-b85ca2667ede.png)

I'm not sure why you put extra spacing on the left of the title. Is it made intentionally? From my point of view, it should be left-aligned as it is right now on the website.

mcturco committed:
Fixing errors

mcturco commented:
Hey @akshatnema, thank you so much for your kind words and your review ❤️ 

> Yep, I completely understand that you have made a default styling for all the links on the website, but I think we have to make the hover effect of this link the same as that of About, Blog, etc. in the footer.

Good find! I have updated the code with your suggestion 😄 

> This error is still on the website right now as well. Looks like there is a very fine gap between the top and navbar of the website. I don't know why it is there and how to resolve it. I will just suggest, if you find this error, kindly resolve it 😄.

I don't see this on my end, even when checking in other browsers (chrome, firefox, safari). Either way, because this is a current problem on the website, we should probably raise a separate issue and debug it there.

> I'm not sure why you put extra spacing on the left of the title. Is it made intentionally? From my point of view, it should be left-aligned as it is right now on the website.

Thank you for finding this bug! That definitely wasn't intentional... not sure how that happened, must have been while merging changes. Just fixed that as well!

magicmatatjahu commented:
Seriously, Missy needs more feedback. it's a big change and we don't want to merge with two reviews. 

cc @derberg @boyney123 @fmvilas @alequetzalli

fmvilas commented:
The roadmap page isn't still visible? Any idea why?

magicmatatjahu commented:
> The roadmap page isn't still visible? Any idea why? 

It is never visible because someone could make a console log of our credentials (I guess, I haven't configured repo config myself).

fmvilas commented:
Oh yeah, you're right. Cool, then I'll blindly say "LGTM 👍" 😄

boyney123 commented:
Scanned through the 45 files of changes, couldn't see any obvious problems, also spent time looking at the preview (which looks amazing btw), can't find any problems either... so from what I can tell it's looking good!

fmvilas committed:
Merge branch 'master' into missy-new-styles

fmvilas committed:
Merge branch 'master' into missy-new-styles

magicmatatjahu reviewed (approved):
LGTM!

quetzalliwrites reviewed (approved):
🚢 🚢 🚢

mcturco commented:
/rtm
